tests/test_pytest_logger.py

```diff
@@ -387,3 +391,87 @@ def test_case{0}():
     for index in range(N):
         logfilename = 'logs/test_case{0}.py/test_case{0}/foo'.format(index)
         FileLineMatcher(basetemp(testdir), logfilename).fnmatch_lines(['* wrn foo: this is test %s' % index])
+
+
+def test_logsdir_option(testdir):
+    makefile(testdir, ['conftest.py'], """
+        import logging
+        def pytest_logger_fileloggers(item):
+            return [
+                'foo',
+                ('bar', logging.ERROR),
+            ]
+    """)
+    makefile(testdir, ['test_case.py'], """
+        import logging
+        def test_case():
+            for lgr in (logging.getLogger(name) for name in ['foo', 'bar']):
+                lgr.fatal('this is fatal')
+                lgr.warning('this is warning')
+    """)
+
+    logsdirname = 'myoptlogs'
```

Comment:
Consider merging both test to single parametrized test (param 1: cmdline, param 2: ini file), as they have 90% of code identical.

Additional checking if directory is purged during 2nd execution would be nice.